tests/Constraints/CoerciveTest.php

```diff
@@ -11,274 +11,181 @@
 
 use JsonSchema\Constraints\Constraint;
 use JsonSchema\Constraints\Factory;
-use JsonSchema\SchemaStorage;
-use JsonSchema\Uri\UriResolver;
+use JsonSchema\Constraints\TypeCheck\LooseTypeCheck;
 use JsonSchema\Validator;
 
-class CoerciveTest extends BasicTypesTest
+class CoerciveTest extends VeryBaseTestCase
 {
-    /**
-     * @dataProvider getValidCoerceTests
-     */
-    public function testValidCoerceCasesUsingAssoc($input, $schema)
-    {
-        $checkMode = Constraint::CHECK_MODE_TYPE_CAST | Constraint::CHECK_MODE_COERCE_TYPES;
-
-        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
-        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
-
-        $validator = new Validator(new Factory($schemaStorage, null, $checkMode));
+    protected $factory = null;
 
-        $value = json_decode($input, true);
-
-        $validator->validate($value, $schema, $checkMode);
-        $this->assertTrue($validator->isValid(), print_r($validator->getErrors(), true));
+    public function setUp()
+    {
+        $this->factory = new Factory();
+        $this->factory->setConfig(Constraint::CHECK_MODE_TYPE_CAST | Constraint::CHECK_MODE_COERCE_TYPES);
     }
 
-    /**
-     * @dataProvider getValidCoerceTests
-     */
-    public function testValidCoerceCases($input, $schema, $errors = array())
+    public function dataCoerceCases()
     {
-        $checkMode = Constraint::CHECK_MODE_TYPE_CAST | Constraint::CHECK_MODE_COERCE_TYPES;
-
-        $schemaStorage = new SchemaStorage($this->getUriRetrieverMock(json_decode($schema)));
-        $schema = $schemaStorage->getSchema('http://www.my-domain.com/schema.json');
+        // check type conversions
+        $types = array(
+            // toType
+            'string' => array(
+                //    fromType      fromValue       toValue         valid   Test Number
+                array('string',     '"string"',     'string',       true),  // #0
```

Comment:
I usually find it confusing when a code sample uses values that mimic or mirror the type they are meant to be demonstrating. Instead of 'string', can we use something else?